File: apps/backend-java/src/main/java/com/enterpriseagentos/backendjava/domain/features/agents/interfaces/IProviderService.java
package com.enterpriseagentos.backendjava.domain.features.agents.interfaces;

import java.util.List;
import java.util.concurrent.CompletableFuture;

import com.enterpriseagentos.backendjava.domain.features.agents.dtos.ProviderDto;

public interface IProviderService {
    CompletableFuture<List<ProviderDto>> listAsync();
    CompletableFuture<String> getApiKeyForDispatchAsync(String name);
}

File: apps/backend/src/EnterpriseAgentOs.Api/Features/Agents/Domain/Interfaces/IProviderService.cs
namespace EnterpriseAgentOs.Domain.Features.Agents;

public interface IProviderService
{
    Task<IReadOnlyList<ProviderDto>> ListAsync(CancellationToken ct = default);

    /// <summary>
    /// Returns the API key for LLM dispatch from environment/config. Returns null if no key is configured.
    /// </summary>
    Task<string?> GetApiKeyForDispatchAsync(string name, CancellationToken ct = default);
}
